MOBT-3-Patterns/MVVMWithoutC/Login/LoginViewModel.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  LoginViewModel.swift
+//  MVVMWithoutC
+//
+//  Created by Mykyta Babanin on 01.07.2021.
+//
+
+import Foundation
+
+
+protocol LoginViewModelProtocol {
+    typealias authenticationLoginCallBack = (_ status: Bool) -> Void
+    var modelUserName: String { get }
+    var modelPassword: String { get }
+    func authenticationUserWith(userName: String, andPassword password: String)
+    func verifyUserWith(_ userName: String, andPassword password: String)
+    func loginCompletion(callBack: @escaping authenticationLoginCallBack)
+    func displaySignUpPage()
+    func displaySignOut()
+    func viewDidDisapear()
+}
+
+class LoginViewModel: LoginViewModelProtocol {
+    weak var coordinator: LoginCoordinator?
+    let defaults: UserDefaults
+    let model: LoginModel
+    
+    var modelUserName: String {
+        get {
+            guard let userName = model.userName else { return "" }
+            return userName }
+    }
+    
+    var modelPassword: String {
+        get {
+            guard let password = model.password else { return "" }
+            return password }
+    }
+    
+    typealias authenticationLoginCallBack = (_ status: Bool) -> Void
+    var loginCallback:authenticationLoginCallBack?
+    
+    init(defaults: UserDefaults = .standard, model: LoginModel) {
+        self.defaults = defaults
+        self.model = model
+    }
+    
+    func authenticationUserWith(userName: String, andPassword password: String) {
+        if userName.count != 0 {
```

Comment:
1) Looks like it would be better to use guard instead of nested if's here
2) Please prefer to use isEmpty instead of count on collections for the [following reasons](https://www.swiftbysundell.com/articles/count-vs-isEmpty/) (unless you need to know the exact count, of course)